src/BluetoothA2DPSource.cpp

```diff
@@ -15,982 +15,1053 @@
 }
 
-void BluetoothA2DPSource::start(const char* name, music_data_channels_cb_t callback) {
-    std::vector<const char*> names = {name};
-    start(names, callback);
+void BluetoothA2DPSource::start(const char *name,
+                                music_data_channels_cb_t callback) {
+  std::vector<const char *> names = {name};
+  start(names, callback);
 }
 
-void BluetoothA2DPSource::start(std::vector<const char*> names, music_data_channels_cb_t callback) {
-    ESP_LOGD(BT_APP_TAG, "%s, ", __func__);
-    if (callback!=NULL){
-        // we use the indicated callback
-        this->data_stream_channels_callback = callback;
-        start_raw(names, ccall_get_channel_data_wrapper);
-    } else {
-        // we use the callback which supports write_data
-        start_raw(names, ccall_get_data_default);
-    }
+void BluetoothA2DPSource::start(std::vector<const char *> names,
+                                music_data_channels_cb_t callback) {
+  ESP_LOGD(BT_APP_TAG, "%s, ", __func__);
+  if (callback != NULL) {
+    // we use the indicated callback
+    this->data_stream_channels_callback = callback;
+    start_raw(names, ccall_get_channel_data_wrapper);
+  } else {
+    // we use the callback which supports write_data
+    start_raw(names, ccall_get_data_default);
+  }
 }
 
-void BluetoothA2DPSource::start_raw(const char* name, music_data_cb_t callback) {
-    std::vector<const char*> names = {name};
-    start_raw(names, callback);
+void BluetoothA2DPSource::start_raw(const char *name,
+                                    music_data_cb_t callback) {
+  std::vector<const char *> names = {name};
+  start_raw(names, callback);
 }
 
-
-void BluetoothA2DPSource::start_raw(std::vector<const char*> names, music_data_cb_t callback) {
-    ESP_LOGD(BT_APP_TAG, "%s, ", __func__);
-    this->bt_names = names;
-    this->data_stream_callback = callback;
-    is_autoreconnect_allowed = true;
-    
-    // get last connection if not available
-    if(!has_last_connection()){
-        get_last_connection();
-    }
-    // reset last connection if we do not reconnect
-    if (reconnect_status==NoReconnect){
-        reset_last_connection();
-    }
-
-    if (nvs_init){
-        // Initialize NVS (Non-volatile storage library).
-        esp_err_t ret = nvs_flash_init();
-        if (ret == ESP_ERR_NVS_NO_FREE_PAGES || ret == ESP_ERR_NVS_NEW_VERSION_FOUND) {
-            ESP_ERROR_CHECK(nvs_flash_erase());
-            ret = nvs_flash_init();
-        }
-        ESP_ERROR_CHECK( ret );
+void BluetoothA2DPSource::start_raw(std::vector<const char *> names,
+                                    music_data_cb_t callback) {
+  ESP_LOGD(BT_APP_TAG, "%s, ", __func__);
+  this->bt_names = names;
+  this->data_stream_callback = callback;
+  is_autoreconnect_allowed = true;
+
+  // get last connection if not available
+  if (!has_last_connection()) {
+    get_last_connection();
+  }
+
+  // reset last connection if we do not reconnect
+  reset_last_connection();
```

Comment:
the if statement was intentionally removed from this?  the comment didn't seem to change